src/server/index.ts

```diff
@@ -131,16 +131,80 @@ export class DocumentDriveServer extends BaseDocumentDriveServer {
         this.initializePromise = this._initialize();
     }
 
-    private updateSyncStatus(
-        driveId: string,
-        status: SyncStatus | null,
+    private getOperationSource(source: StrandUpdateSource) {
+        return source.type === 'local' ? 'push' : 'pull';
+    }
+
+    private getCombinedSyncUnitStatus(
+        syncUnitStatus: SyncUnitStatusObject
+    ): SyncUnitStatus {
+        if (syncUnitStatus.pull === 'INITIAL_SYNC') return 'INITIAL_SYNC';
+        if (syncUnitStatus.push === 'INITIAL_SYNC') return syncUnitStatus.pull;
```

Comment:
I think this should be:
```
if (syncUnitStatus.push === 'INITIAL_SYNC') return syncUnitStatus.pull || 'INITIAL_SYNC';
```
In case the unit is being pushed but not pulled.